pkg/globalnet/controllers/gateway_monitor.go

```diff
@@ -353,9 +439,10 @@ func (g *gatewayMonitor) startControllers() error {
 	return nil
 }
 
-func (g *gatewayMonitor) stopControllers() {
+func (g *gatewayMonitor) stopControllers(timeout int) {
```

Comment:
```suggestion
func (g *gatewayMonitor) stopControllers(leaderElectionTimeout int) {
```

but that’s not great, because the timeout isn’t used for the election timeout itself, but when stopping the controller; so perhaps `leaderElectionControllerTimeout` but that’s a mouthful.

Reply:
The timeout handling here might be better done with a context with a timeout ;-). (Just a thought, I haven’t tried it.)

Reply:
I changed it to a context.